coordinator/src/services/translation/incoming/message-header.ts

```diff
@@ -4,31 +4,36 @@ import {createIdentifier, createReference} from "./fhir-base-types"
 
 export function createMessageHeader(
   messageId: string,
-  patientReference: string,
-  medicationRequestReference: string,
-  representedOrganizationReference: string,
-  cancelRequestId: string
+  eventCoding: fhir.Coding,
+  focusIds: Array<string>,
+  destinationOrganizationId: string,
+  requestMessageId: string
 ): fhir.MessageHeader {
   return {
     resourceType: "MessageHeader",
     id: generateResourceId(),
     extension: getExtensions(messageId),
-    eventCoding: getEventCoding(),
-    destination: getDestinations(representedOrganizationReference),
+    eventCoding: eventCoding,
+    destination: getDestinations(destinationOrganizationId),
     sender: getNhsdSender(),
     source: getSource(),
-    response: getMessageHeaderResponse(cancelRequestId),
-    focus: createFocus(patientReference, medicationRequestReference)
+    response: getMessageHeaderResponse(requestMessageId),
+    focus: createFocus(focusIds)
   }
 }
 
-function getEventCoding() {
-  return {
+export const EVENT_CODING: Record<string, fhir.Coding> = Object.freeze({
```

Comment:
Not seen Object.freeze before. Do we really need it? Is it just to stop things getting changed accidentally further downstream?

Reply:
Yeah just makes the entire thing immutable. Don't _need_ it, but I think it's good practice

Reply:
Might get rid of it, what do other people think?

Reply:
In this case it looks like we'd need it to deep freeze to make it useful (recursively go over all properties as we have nested objects. Might be overkill and just slightly unnecessary added complexity in readability, especially if we have tests covering the output anway?